rasa/core/exporter.py

```diff
@@ -279,44 +276,3 @@ def _get_events_for_conversation_id(
             events_with_conversation_id.append(event)
 
         return events_with_conversation_id
-
-    def _sort_and_select_events_by_timestamp(
-        self, events: Iterable[Dict[Text, Any]]
-    ) -> List[Dict[Text, Any]]:
-        """Sort list of events by ascending timestamp, select events within time range.
-
-        Args:
-            events: List of serialized events to be sorted and selected from.
-
-        Returns:
-            List of serialized and sorted (by timestamp) events within the requested
-            time range.
-
-        Raises:
-             `NoEventsInTimeRangeError` error if no events are found within the
-             requested time range.
-
-        """
-        logger.debug(f"Sorting and selecting from {len(events)} total events found.")
-        # sort the events by timestamp just in case they're not sorted already
-        events = sorted(events, key=lambda x: x["timestamp"])
-
-        # drop events failing minimum timestamp requirement
-        if self.minimum_timestamp is not None:
-            events = itertools.dropwhile(
-                lambda x: x["timestamp"] < self.minimum_timestamp, events
-            )
-
-        # select events passing maximum timestamp requirement
-        if self.maximum_timestamp is not None:
-            events = itertools.takewhile(
-                lambda x: x["timestamp"] < self.maximum_timestamp, events
-            )
-
-        events = list(events)
-        if not events:
-            raise NoEventsInTimeRangeError(
-                "Could not find any events within requested time range. Exiting."
-            )
```

Comment:
Should we retain raising this exception if `published_events == 0` in `publish_events()` after going through generator results?

Reply:
I don't think this is helpful tbh. There is output on how many events got produced and that will just be 0 in this case. but happy to readd it

Reply:
Makes sense! I asked in case this exception is handled maybe in some custom way either in the codebase already or in a customer's project, that could have required further amendments.